scripts/zones/RoMaeve/Zone.lua

```diff
@@ -35,21 +35,43 @@ function onRegionEnter(player, region)
 end
 
 function onGameHour(zone)
-    local hour = VanadielHour()
-
-    if IsMoonFull() and hour == 18 then
-        for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET+7 do
-            GetNPCByID(i):openDoor(1728)
-            GetNPCByID(i):untargetable(true)
+    local moongate = GetNPCByID(ID.npc.MOONGATE_OFFSET)
+    local romaeveActive = moongate:getLocalVar("romaeveActive")
+    
+    -- Make Ro'Maeve come to life between 6pm and 6am during a full moon
+    if
+        romaeveActive == 0 and IsMoonFull() and
+        (VanadielHour() >= 18 or VanadielHour() < 6)
+    then
+        -- Calculate remaining seconds till 6am from this hour
+        local secondsTillMorning
+        if VanadielHour() < 6 then
```

Comment:
We really should keep the variable rather than repeatedly call the same function. OnGameHour is going to re execute every hour, so the results isn’t able to change before reaching the next place it is checked

Reply:
It won't be called repeatedly at all since we set `setLocalVar("romaeveActive", 1)` though? none of all of that will be called again after firing once.

Reply:
you have a condition inside of a condition that both ask what the hour is. you can store the function call result instead of calling it twice.

Reply:
I commented at the line where the 2nd one happens - I'm not talking a bout your loop. But while we're at it, why are we looping for hrs left instead of using >= and <= since we know the exact hr ranges on both sides of 0:00 and 24:00

Reply:
Oh yeah, that's a different thing. Makes sense.

Reply:
if not active and full moon and hour <6  then.. If hour <6  :thinking:

Reply:
On hold for now after talking to @TeoTwawki some more.